flink-tests/src/test/java/org/apache/flink/test/runtime/leaderelection/ZooKeeperLeaderElectionITCase.java

```diff
@@ -1,301 +0,0 @@
-import akka.actor.PoisonPill;
-import org.apache.curator.test.TestingServer;
-import org.junit.AfterClass;
-import org.junit.BeforeClass;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.TemporaryFolder;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import java.io.File;
-import java.util.UUID;
-
-import scala.concurrent.Await;
-import scala.concurrent.Future;
-import scala.concurrent.duration.Deadline;
-import scala.concurrent.duration.FiniteDuration;
-import scala.concurrent.impl.Promise;
-
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.fail;
-
-/**
- * Test the election of a new JobManager leader.
- */
-public class ZooKeeperLeaderElectionITCase extends TestLogger {
-
-	private static final FiniteDuration timeout = TestingUtils.TESTING_DURATION();
-
-	private static TestingServer zkServer;
-
-	@Rule
-	public TemporaryFolder tempFolder = new TemporaryFolder();
-
-	@BeforeClass
-	public static void setup() throws Exception {
-		zkServer = new TestingServer(true);
-	}
-
-	@AfterClass
-	public static void tearDown() throws Exception {
-		if (zkServer != null) {
-			zkServer.close();
-			zkServer = null;
-		}
-	}
-
-	/**
-	 * Tests that the TaskManagers successfully register at the new leader once the old leader
-	 * is terminated.
-	 */
-	@Test
-	public void testTaskManagerRegistrationAtReelectedLeader() throws Exception {
-		File rootFolder = tempFolder.getRoot();
-
-		Configuration configuration = ZooKeeperTestUtils.createZooKeeperHAConfig(
-			zkServer.getConnectString(),
-			rootFolder.getPath());
-		configuration.setString(HighAvailabilityOptions.HA_CLUSTER_ID, UUID.randomUUID().toString());
-
-		int numJMs = 10;
-		int numTMs = 3;
-
-		configuration.setInteger(ConfigConstants.LOCAL_NUMBER_JOB_MANAGER, numJMs);
-		configuration.setInteger(ConfigConstants.LOCAL_NUMBER_TASK_MANAGER, numTMs);
-
-		TestingCluster cluster = new TestingCluster(configuration);
-
-		try {
-			cluster.start();
-
-			for (int i = 0; i < numJMs; i++) {
-				ActorGateway leadingJM = cluster.getLeaderGateway(timeout);
-
-				cluster.waitForTaskManagersToBeRegisteredAtJobManager(leadingJM.actor());
-
-				Future<Object> registeredTMs = leadingJM.ask(
-					JobManagerMessages.getRequestNumberRegisteredTaskManager(),
-					timeout);
-
-				int numRegisteredTMs = (Integer) Await.result(registeredTMs, timeout);
-
-				assertEquals(numTMs, numRegisteredTMs);
-
-				cluster.clearLeader();
-				leadingJM.tell(PoisonPill.getInstance());
-			}
-		} finally {
-			cluster.stop();
-		}
-	}
-
-	/**
-	 * Tests that a job can be executed after a new leader has been elected. For all except for the
-	 * last leader, the job is blocking. The JobManager will be terminated while executing the
-	 * blocking job. Once only one JobManager is left, it is checked that a non-blocking can be
-	 * successfully executed.
-	 */
-	@Test
-	public void testJobExecutionOnClusterWithLeaderReelection() throws Exception {
```

Comment:
I'm not sure whether we can simply drop this test because it tests a bit more than that all components will reconnect to the leader. It also tests that the `Dispatcher` will recover the submitted job from the `ZooKeeperSubmittedJobGraphStore` and then executes this job. I think it still makes sense to keep this test.